### PR TITLE
[7836]Fix Cannot create API token via API (hen and egg)

### LINKED ISSUE #7836: Cannot create API token via API (hen and egg)
## Describe the bug
CKAN 2.9 supported API keys which made it possible to create a user via the API (`ckan.auth.create_user_via_api = true`) and automatically return the user's API key (`ckan.auth.create_default_api_keys = true`).

Since CKAN 2.10 does not support API keys, it is not possible anymore to use the API anonymously to create users and start working as them. An API token is required to create an API token for a user. So users would have to log in via the web interface first to create an API token.

### PR DESCRIPTION
**Pull request by TomeCirun**
Fixes #7836 

### Features:

- [x] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by TomeCirun** [comment]
@smotornyuk done

File: ckan/logic/action/create.py [hunk 1]
```diff
@@ -1032,10 +1035,18 @@ def user_create(context: Context,
...
     dashboard = model.Dashboard(user.id)
+
     session.add(dashboard)
     if not context.get('defer_commit'):
         model.repo.commit()
 
+    if with_apitoken:
+        # Create apitoken for user.
+        api_token = _get_action("api_token_create")(
+            context, {"user": user.name, "name": "default"}
```
Comment (smotornyuk) [inline ckan/logic/action/create.py:1046]: It will cause an AuthorizationError because you are potentially using anonymous context

**Comment by smotornyuk** [comment]
Found a problem - anonymous user will get an error during api request

File: ckan/logic/action/create.py (hunk 1 above)
Comment (TomeCirun) [reply ckan/logic/action/create.py:1049]: @smotornyuk done

**Pull request merged**